x-pack/plugins/security_solution/public/common/components/user_profiles/use_get_current_user_profile.tsx

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useQuery } from '@tanstack/react-query';
+
+import type { SecurityPluginStart } from '@kbn/security-plugin/public';
+import type { UserProfileWithAvatar } from '@kbn/user-profile-components';
+
+import { CURRENT_USER_PROFILE_FAILURE } from './translations';
+import { useKibana } from '../../lib/kibana';
+import { useAppToasts } from '../../hooks/use_app_toasts';
+
+export const getCurrentUserProfile = async ({
+  security,
+}: {
+  security: SecurityPluginStart;
+}): Promise<UserProfileWithAvatar> => {
+  return security.userProfiles.getCurrent({ dataPath: 'avatar' });
+};
+
+export const useGetCurrentUserProfile = () => {
```

Comment:
Nit: can you also add a note of when to use this hook vs the other current user hooks that exist?

Reply:
done